nexus/src/app/background/tasks/support_bundle_collector.rs

```diff
@@ -981,6 +1007,85 @@ where
     Ok(())
 }
 
+/// Collect task dumps from all SPs via MGS and save them to a directory.
+async fn save_all_sp_dumps(
+    log: &slog::Logger,
+    resolver: &Resolver,
+    sp_dumps_dir: &Utf8Path,
+) -> anyhow::Result<()> {
+    let mgs_client = resolver
+        .lookup_socket_v6(ServiceName::ManagementGatewayService)
+        .await
+        .map(|sockaddr| {
+            let url = format!("http://{}", sockaddr);
+            gateway_client::Client::new(&url, log.clone())
+        })
+        .context("failed to resolve address of MGS")?;
+
+    let all_sps = mgs_client
+        .sp_all_ids()
+        .await
+        .context("failed to get list of SPs from MGS")?
+        .into_inner();
+
+    let mut futures = futures::stream::iter(all_sps.into_iter())
+        .map(|sp| {
+            let mgs_client = mgs_client.clone();
+
+            async move {
+                save_sp_dumps(mgs_client, sp, &sp_dumps_dir)
+                    .await
+                    .with_context(|| format!("SP {} {}", sp.type_, sp.slot))
+            }
+        })
+        .buffer_unordered(10);
+
+    while let Some(result) = futures.next().await {
+        if let Err(e) = result {
+            error!(
+                log,
+                "failed to capture task dumps";
+                "error" => InlineErrorChain::new(e.as_ref())
+            );
+        }
+    }
+
+    Ok(())
+}
+
+/// Fetch and save task dumps from a single SP.
+async fn save_sp_dumps(
+    mgs_client: MgsClient,
+    sp: SpIdentifier,
+    sp_dumps_dir: &Utf8Path,
+) -> anyhow::Result<()> {
+    let dump_count = mgs_client
+        .sp_task_dump_count(sp.type_, sp.slot)
```

Comment:
When / how are these task dumps deleted? Just trying to understand if there's a TOCTTOU issue between "get # of tasks" vs "iterate over them"